tests/D2L.CodeStyle.Analyzers.Test/Specs/SerializerAttributeAnalyzer.cs

```diff
@@ -0,0 +1,39 @@
+﻿// analyzer: D2L.CodeStyle.Analyzers.ApiUsage.Serialization.SerializerAttributeAnalyzer
+
+using System;
+using System.ComponentModel;
+using D2L.LP.Serialization;
+
+namespace D2L.LP.Serialization {
+	public sealed class SerializerAttribute : Attribute {
+		public SerializableAttribute( Type type ) { }
+	}
+	public interface ITrySerializer { }
+}
+
+namespace SpecTests {
+
+	[Serializer( typeof( Nested.Serializer ) )]
+	public sealed class Nested {
+		private sealed class Serializer : ITrySerializer { }
+	}
+
+	[Serializer( typeof( ExternalSerializer ) )]
+	public sealed class External { }
+	public sealed class ExternalSerializer : ITrySerializer { }
+
+	[Serializer( /* InvalidSerializerType(System.String) */ typeof( string ) /**/ )]
+	public sealed class InvalidSerializerType { }
+
+	[Serializer( /* InvalidSerializerType(null) */ null /**/ )]
+	public sealed class NullSerializerType { }
+
+	[Serializer( /* InvalidSerializerType(123) */ 123 /**/ )]
+	public sealed class InvalidSerializerTypeArgument { }
+
+	[Serializer( )]

```

Comment:
Could you add a case for `[Serializer]` as well? It might produce an NRE on accessing `.ArgumentList`.

Reply:
Good catch.  `ArgumentList` is `null` in this case